scripts/Valves/valve.py

```diff
@@ -0,0 +1,448 @@
+    x4 = x2 + R * math.cos((math.pi / 2.0) - alpha)
+    y4 = y2 + R * math.sin((math.pi / 2.0) - alpha)
+  
+    #
+    th = math.sqrt((((0.0 - sadle_x) - x3)**2) + ((0.0 - y3)**2))
+    aa = math.fabs(math.degrees(math.acos((x3 - (0.0 - sadle_x)) / th))) - 90.0
+    #
+    # reflect Y in small circle
+    #
+    y5 = 0.0 - (y3 - 0.0)
+    y6 = 0.0 - (y4 - 0.0)
+    #
+    x5 = x3
+    x6 = x4
+    # Upper left line
+    #
+    # Upper part of large circle
+    th = math.sqrt(((0.0 - x6)**2) + ((0.0 - y6)**2))
+    aa2 = 90.0 - math.fabs(math.degrees(math.acos((0.0 - x6) / th)))
+    #
+    # reflect X in large circle
+    #
+    x7 = 0.0 + (0.0 - x6)
+    y7 = y6
+    #
+    # reflect small circle in the large circles origo
+    #
+    x8 = 0.0 + (0.0 - x5)
+    y8 = y5
+    #
+    #
+    # reflect small circle in the large circles origo
+    #
+    #
+    # reflect large circle in the large circles origo
+    #
+    x10 = x7
+    y10 = 0.0 + (0.0 - y7)
+    x9 = x8
+    y9 = y3
+    #
+    # Rotate all postions
+    #
+    x3n = x3 * math.cos(math.radians(sadle_a)) - (y3 * math.sin(math.radians(sadle_a)))
+    y3n = x3 * math.sin(math.radians(sadle_a)) + (y3 * math.cos(math.radians(sadle_a)))
+    #
+    x4n = x4 * math.cos(math.radians(sadle_a)) - (y4 * math.sin(math.radians(sadle_a)))
+    y4n = x4 * math.sin(math.radians(sadle_a)) + (y4 * math.cos(math.radians(sadle_a)))
+    #
+    xL1n = (0 - sadle_x) * math.cos(math.radians(sadle_a)) - (0.0 * math.sin(math.radians(sadle_a)))
+    yL1n = (0 - sadle_x) * math.sin(math.radians(sadle_a)) + (0.0 * math.cos(math.radians(sadle_a)))
+    #
+    x5n = x5 * math.cos(math.radians(sadle_a)) - (y5 * math.sin(math.radians(sadle_a)))
+    y5n = x5 * math.sin(math.radians(sadle_a)) + (y5 * math.cos(math.radians(sadle_a)))
+    #
+    x6n = x6 * math.cos(math.radians(sadle_a)) - (y6 * math.sin(math.radians(sadle_a)))
+    y6n = x6 * math.sin(math.radians(sadle_a)) + (y6 * math.cos(math.radians(sadle_a)))
+    #
+    #
+    #
+    x7n = x7 * math.cos(math.radians(sadle_a)) - (y7 * math.sin(math.radians(sadle_a)))
+    y7n = x7 * math.sin(math.radians(sadle_a)) + (y7 * math.cos(math.radians(sadle_a)))
+    #
+    x8n = x8 * math.cos(math.radians(sadle_a)) - (y8 * math.sin(math.radians(sadle_a)))
+    y8n = x8 * math.sin(math.radians(sadle_a)) + (y8 * math.cos(math.radians(sadle_a)))
+    #
+    xL2n = sadle_x * math.cos(math.radians(sadle_a)) - (0.0 * math.sin(math.radians(sadle_a)))
+    yL2n = sadle_x * math.sin(math.radians(sadle_a)) + (0.0 * math.cos(math.radians(sadle_a)))
+    #
+    #
+    #
+    x9n = x9 * math.cos(math.radians(sadle_a)) - (y9 * math.sin(math.radians(sadle_a)))
+    y9n = x9 * math.sin(math.radians(sadle_a)) + (y9 * math.cos(math.radians(sadle_a)))
+    #
+    x10n = x10 * math.cos(math.radians(sadle_a)) - (y10 * math.sin(math.radians(sadle_a)))
+    y10n = x10 * math.sin(math.radians(sadle_a)) + (y10 * math.cos(math.radians(sadle_a)))
```

Comment:
all of these rotations would be a lot simpler if using an array and either a rotate function or maybe even better merge https://github.com/pointhi/kicad-footprint-generator/pull/351 into your branch as it defines a rotation function for Vector2D.

So the options are:
```python
# somewhere near the top of the file
def rotate(x, y, angle):
  xo = x * math.cos(math.radians(angle)) - (y * math.sin(math.radians(angle)))
  yo = x * math.sin(math.radians(angle)) + (y * math.cos(math.radians(angle)))
  return xo, yo

# coords_0 are Vector2D
coords = []
for v in coords_0:
  coords.append(Vector2D(rotate(v.x, v.y, sadle_a)))
```
---

Or if you merged my stuff then:
```python
coords = []
for v in coords_0:
  coords.append(Vector2D(v).rotate(sadle_a))
```
or if you do not need the orriginal coordinates then it could look like this:
```python
for i in range(len(coords)):
  coords[i].rotate(sadle_a)
```